### PR TITLE
feat: remove the created intermediate containers while building the lambda functions of Image Type

### PR DESCRIPTION
**Pull request by moelasmar**
#### Which issue(s) does this change fix?
The default behaviour of [docker image create API](https://docker-py.readthedocs.io/en/stable/images.html#docker.models.images.ImageCollection.build) is to keep the intermediate containers created. So, when customer execute `sam build` for an Image type Lambda Function, the customer will find some created containers exist after `sam build` command finish. This PR is to clean these intermediate containers, and let the docker API delete these containers after the image built process finish.

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Comment by jfuss** [comment]
Is this the right behavior we want? I ask because from https://docs.docker.com/engine/reference/commandline/images/: "Docker images have intermediate layers that increase reusability, decrease disk usage, and speed up docker build by allowing each step to be cached. These intermediate layers are not shown by default." This suggests that (at least currently) some customers may actually want this as the behavior because it saves the layers on the machine to be re-used. And I assume this leads to faster builds, if you build twice.

**Comment by moelasmar** [comment]
my understanding is this change will only remove the intermediate containers, but will keep the images. This is the same behaviour we implement in `sam build --use-container`, we cleaned the created containers after build is done.

By the way, this also the same behaviour of `docker build` command.

File: samcli/lib/build/app_builder.py [hunk 1]
```diff
@@ -413,6 +413,8 @@ def _build_lambda_image(self, function_name: str, metadata: Dict, architecture:
             "buildargs": docker_build_args,
             "decode": True,
             "platform": get_docker_platform(architecture),
+            "rm": True,
+            "forcerm": True,
```
Comment (torresxb1) [inline samcli/lib/build/app_builder.py:417]: I see that here we have `rm` True but not `forcerm`: https://github.com/aws/aws-sam-cli/blob/develop/samcli/local/docker/lambda_image.py#L266
Should we keep them consistent?

File: tests/integration/buildcmd/test_build_cmd.py [hunk 2]
```diff
@@ -46,11 +46,6 @@
 SKIP_SAR_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
 
 
-@skipIf(
-    # Hits public ECR pull limitation, move it to canary tests
-    ((not RUN_BY_CANARY) or (IS_WINDOWS and RUNNING_ON_CI) and not CI_OVERRIDE),
-    "Skip build tests on windows when running in CI unless overridden",
-)
```
Comment (torresxb1) [inline tests/integration/buildcmd/test_build_cmd.py:53]: we don't want to skip anymore?

**Review by torresxb1 — COMMENTED** [review]
couple of questions

File: tests/integration/buildcmd/test_build_cmd.py (hunk 2 above)
Comment (moelasmar) [reply tests/integration/buildcmd/test_build_cmd.py:53]: no, it is my bad .. I removed them to run testing in my local machine, and forget to return them back :(

File: samcli/lib/build/app_builder.py (hunk 1 above)
Comment (moelasmar) [reply samcli/lib/build/app_builder.py:417]: actually I was hesitated to add `forcerm`, I will remove it.

**Review by jfuss — APPROVED** [review]

**Review by torresxb1 — APPROVED** [review]

**Pull request merged**